python/cuml/cuml/model_selection/_split.py

```diff
@@ -419,7 +421,106 @@ def _process_df_objs(
         return X_train, X_test
 
 
-class StratifiedKFold:
+class _KFoldBase(ABC):
+    """Base class for k-fold split."""
+
+    def __init__(self, n_splits=5, shuffle=False, random_state=None):
+        if n_splits < 2 or not isinstance(n_splits, int):
+            raise ValueError(
+                f"n_splits {n_splits} is not a integer at least 2"
+            )
+
+        self.n_splits = n_splits
+        self.shuffle = shuffle
+        self.seed = random_state
+
+    @abstractmethod
+    def split(self, x, y):
+        """Generate indices to split data into training and test set.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,), default=None
+            The target variable for supervised learning problems.
+
+        Yields
+        ------
+        train_idx : CuPy ndarray
+            The training set indices for that split.
+
+        test_idx : CuPy ndarray
+            The testing set indices for that split.
+        """
+        raise NotImplementedError()
+
+
+class KFold(_KFoldBase):
```

Comment:
Why are we not supporting the `get_n_splits` method?

Reply:
added